provider/sdk/dkubefs/dkube_store.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+from feast import RepoConfig
+from feast.entity import Entity
+from feast.feature_table import FeatureTable
+from feast.feature_view import FeatureView
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel
+# from mysql.connector import connect
+from pydantic import StrictStr
+from pydantic.typing import Literal
+from provider.sdk.dkubefs.online_drivers.local_driver import OnlineLocalDriver
+
+from provider.sdk.dkubefs.online_drivers.remote_driver import OnlineRemoteDriver
+
+
+class DkubeOnlineStoreConfig(FeastConfigBaseModel):
+    type: Literal[
+        "dkubefs.dkube_store.DkubeOnlineStore"
+    ] = "dkubefs.dkube_store.DkubeOnlineStore"
+
+
+class DkubeOnlineStore(OnlineStore):
+    driver: Union[OnlineRemoteDriver, OnlineLocalDriver] = None
+
+    def initialize(self, config):
+        if self.driver:
+            return self.driver
+        self.driver = {}
+        self.driver["remote"] = OnlineRemoteDriver(config)
+        self.driver["local"] = OnlineLocalDriver(config)
+
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        data: List[
+            Tuple[
+                EntityKeyProto,
+                Dict[str, ValueProto],
+                datetime,
+                Optional[datetime],
+            ]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        self.initialize(config)
+        self.driver["remote"].online_write_batch(config, table, data, progress)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        entity_keys: List[EntityKeyProto],
+        requested_features: List[str] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        self.initialize(config)
+        return self.driver["local"].online_read(config, table, entity_keys, requested_features)
+
+    def update(
+        self,
+        config: RepoConfig,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
```

Comment:
If there is new FV definition, the older one with registry will get deleted and the new definition will get stored. If the table is updated it will be part of table_to_keep.